src/cosmos/CosmosEngine.ts

```diff
@@ -192,8 +197,22 @@ export class CosmosEngine extends CurrencyEngine<
 
   async queryTransactions(): Promise<void> {
     let progress = 0
+    const { stargateClient, tendermintClient } =
+      Date.now() - TWO_WEEKS > this.otherData.archivedTxLastCheckTime
+        ? // Uses archive rpc for first sync and then only if it's been two weeks between syncs.
+          await createStargateClient(
+            this.fetchCors,
+            this.networkInfo.archiveNode
+          )
+        : // Otherwise, uses regular rpc
+          this.getClients()
+
     for (const query of txQueryStrings) {
-      const newestTxid = await this.queryTransactionsInner(query)
+      const newestTxid = await this.queryTransactionsInner(
+        query,
```

Comment:
Just pass the full `CosmosClients` to `queryTransactionsInner` and remove the destructuring

Reply:
done